server/src/controllers/score.controller.js

```diff
@@ -1,21 +1,33 @@
 import constants from '../utils/contants';
 import {
   queryScores,
+  queryScoresCount,
+  queryRank,
   queryUserScore,
   insertScore,
 } from '../utils/db_score_query';
 
-const {
-  HTTP_SUCCESS_OK,
-  HTTP_INTERNAL_SERVER_ERROR,
-  HTTP_CREATED,
-} = constants;
+const { HTTP_SUCCESS_OK, HTTP_INTERNAL_SERVER_ERROR, HTTP_CREATED } = constants;
 
 const getScores = async (req, res) => {
-  const { range } = req.params;
+  const { offset, limit } = req.query;
   try {
-    const scores = await queryScores(parseInt(range, 10));
-    res.status(HTTP_SUCCESS_OK).json(scores);
+    const scores = queryScores(parseInt(offset, 10), parseInt(limit, 10));
+    const totalScores = queryScoresCount();
+    const data = await Promise.all([scores, totalScores]);
+    res.status(HTTP_SUCCESS_OK).json(data);
+  } catch (error) {
+    res.status(HTTP_INTERNAL_SERVER_ERROR).json({
+      error,
+    });
+  }
+};
+
+const getUserRank = async (req, res) => {
+  const { userId } = req.query;
+  try {
+    const rank = await queryRank(userId);
+    res.status(HTTP_SUCCESS_OK).json(rank);
   } catch (error) {
```

Comment:
Create a wrapper around it, so you don't have to send the same status, error every time like
```suggestion
 const sendResultOrError = async ({ query, res, successStatus = HTTP_SUCCESS_OK, errorStatus = HTTP_INTERNAL_SERVER_ERROR}) => {
   try {
       const result = await query;
       res.status(successStatus).json(result);
   } catch (error) {
       res.status(errorStatus).json({ error })
   }
 }
```
Then you can use with every api call like
```suggestion
  return sendResultOrError({ query: queryRank(userId), res })
```

Reply:
@nimish-gupta That seems like a really good way to refactor.
- But I think that'd increase complexity as we've only two controllers having the same functionality.
- I think the code is more readable now than changes you suggest.
- As, we'd have to look another function to understand the controller functionality just for two controllers out of total four.
- According to me, the changes you suggest would make more sense when we've more than two controllers having same functionality.

I maybe wrong, Please let me know your thoughts on this :)

Reply:
This will be not specific to score controller only but it can be used with all the request's responses.

Reply:
But, there are also other controller specific things going on in the `try block`. How can I wrap it then?
- Do you mean do all those stuff in a `try` block and call this `sendResultOrError` function in that try block?

Reply:
```suggestion

const sendResultOrError = async ({ query, res, errorStatus = HTTP_INTERNAL_SERVER_ERROR}) => {
   try {
       const result = await query;
       return result;
   } catch (error) {
       res.status(errorStatus).json({ error })
   }
 }
```